### ISSUE TITLE
Alter table drop partition returns: No active replicas error

### ISSUE THREAD

**Issue by cw9** [ClickHouse/ClickHouse#10264]
**Describe the bug**
Alter table drop partition returns `DB::Exception: No active replicas (version 20.1.9.54 (official build))`

**How to reproduce**
* Which ClickHouse server version to use
20.1.9.54, but we recently downgraded from 20.3.5, those tables maybe built with version 20.3.5

* Queries to run that lead to unexpected result
ALTER TABLE `ReplicatedMergeTree.foo` DROP PARTITION 1234

**Expected behavior**
Partition should be dropped without error

**Error message and/or stacktrace**
```
{"log":"2020.04.14 19:51:11.076925 [ 641 ] {e23c0d80-b41f-4404-b4d1-17d7a87e4882} \u003cError\u003e executeQuery: Code: 254, e.displayText() = DB::Exception: No active replicas (version 20.1.9.54 (official build)) (from 10.82.212.100:38208) (in query: ALTER TABLE `ReplicatedMergeTree.foo` DROP PARTITION 1234 ), Stack trace (when copying this message, always include the lines below):\n"}
{"log":"\n"}
{"log":"0. 0x10335e20 Poco::Exception::Exception(std::__1::basic_string\u003cchar, std::__1::char_traits\u003cchar\u003e, std::__1::allocator\u003cchar\u003e \u003e const\u0026, int) /build/obj-x86_64-linux-gnu/../contrib/poco/Foundation/src/Exception.cpp:27 in /usr/lib/debug/usr/bin/clickhouse\n"}
{"log":"1. 0x8ea65cd DB::Exception::Exception(std::__1::basic_string\u003cchar, std::__1::char_traits\u003cchar\u003e, std::__1::allocator\u003cchar\u003e \u003e const\u0026, int) /build/obj-x86_64-linux-gnu/../dbms/src/Common/Exception.cpp:35 in /usr/lib/debug/usr/bin/clickhouse\n"}
{"log":"2. 0xd6b7516 DB::StorageReplicatedMergeTree::sendRequestToLeaderReplica(std::__1::shared_ptr\u003cDB::IAST\u003e const\u0026, DB::Context const\u0026) (.cold) /build/obj-x86_64-linux-gnu/../contrib/libcxx/include/string:2134 in /usr/lib/debug/usr/bin/clickhouse\n"}
{"log":"3. 0xd6a2c3e DB::StorageReplicatedMergeTree::dropPartition(std::__1::shared_ptr\u003cDB::IAST\u003e const\u0026, std::__1::shared_ptr\u003cDB::IAST\u003e const\u0026, bool, DB::Context const\u0026) /build/obj-x86_64-linux-gnu/../dbms/src/Storages/StorageReplicatedMergeTree.cpp:3689 in /usr/lib/debug/usr/bin/clickhouse\n"}
{"log":"4. 0xd6a655e DB::StorageReplicatedMergeTree::alterPartition(std::__1::shared_ptr\u003cDB::IAST\u003e const\u0026, DB::PartitionCommands const\u0026, DB::Context const\u0026) /build/obj-x86_64-linux-gnu/../dbms/src/Storages/StorageReplicatedMergeTree.cpp:3521 in /usr/lib/debug/usr/bin/clickhouse\n"}
{"log":"5. 0xd070e5c DB::InterpreterAlterQuery::execute() /build/obj-x86_64-linux-gnu/../contrib/libcxx/include/vector:662 in /usr/lib/debug/usr/bin/clickhouse\n"}
{"log":"6. 0xd48f5e4 DB::executeQueryImpl(char const*, char const*, DB::Context\u0026, bool, DB::QueryProcessingStage::Enum, bool, DB::ReadBuffer*, bool) /build/obj-x86_64-linux-gnu/../dbms/src/Interpreters/executeQuery.cpp:329 in /usr/lib/debug/usr/bin/clickhouse\n"}
{"log":"7. 0xd4920e6 DB::executeQuery(std::__1::basic_string\u003cchar, std::__1::char_traits\u003cchar\u003e, std::__1::allocator\u003cchar\u003e \u003e const\u0026, DB::Context\u0026, bool, DB::QueryProcessingStage::Enum, bool, bool) /build/obj-x86_64-linux-gnu/../dbms/src/Interpreters/executeQuery.cpp:574 in /usr/lib/debug/usr/bin/clickhouse\n"}
{"log":"8. 0x8f96f79 DB::TCPHandler::runImpl() /build/obj-x86_64-linux-gnu/../dbms/programs/server/TCPHandler.cpp:248 in /usr/lib/debug/usr/bin/clickhouse\n"}
{"log":"9. 0x8f97f70 DB::TCPHandler::run() /build/obj-x86_64-linux-gnu/../dbms/programs/server/TCPHandler.cpp:1251 in /usr/lib/debug/usr/bin/clickhouse\n"}
{"log":"10. 0xe1fbf6b Poco::Net::TCPServerConnection::start() /build/obj-x86_64-linux-gnu/../contrib/poco/Net/src/TCPServerConnection.cpp:57 in /usr/lib/debug/usr/bin/clickhouse\n"}
{"log":"11. 0xe1fc3ed Poco::Net::TCPServerDispatcher::run() /build/obj-x86_64-linux-gnu/../contrib/libcxx/include/atomic:856 in /usr/lib/debug/usr/bin/clickhouse\n"}
{"log":"12. 0x103c3f87 Poco::PooledThread::run() /build/obj-x86_64-linux-gnu/../contrib/poco/Foundation/include/Poco/Mutex_STD.h:132 in /usr/lib/debug/usr/bin/clickhouse\n"}
{"log":"13. 0x103bfd8c Poco::ThreadImpl::runnableEntry(void*) /build/obj-x86_64-linux-gnu/../contrib/poco/Foundation/include/Poco/AutoPtr.h:205 in /usr/lib/debug/usr/bin/clickhouse\n"}
{"log":"14. 0x103c172d void* std::__1::__thread_proxy\u003cstd::__1::tuple\u003cstd::__1::unique_ptr\u003cstd::__1::__thread_struct, std::__1::default_delete\u003cstd::__1::__thread_struct\u003e \u003e, void* (*)(void*), Poco::ThreadImpl*\u003e \u003e(void*) /build/obj-x86_64-linux-gnu/../contrib/libcxx/include/memory:2615 in /usr/lib/debug/usr/bin/clickhouse\n"}
{"log":"15. 0x8064 start_thread  in /lib/x86_64-linux-gnu/libpthread-2.19.so\n"}
{"log":"16. 0xe862d __clone  in /lib/x86_64-linux-gnu/libc-2.19.so\n"}
{"log":"\n"}
```

**Additional context**
The data parts are all active according to system.parts table
```

SELECT active
FROM system.parts
WHERE (partition = '1234') AND (table = 'foo')

┌─active─┐
│      1 │
│      1 │
│      1 │
│      1 │
│      1 │
└────────┘
```

**Comment by alesapin** [comment]
Let's look at `SELECT * FROM system.replcas WHERE table = 'foo'`. Also information from zk nodes `/path_to_table/leader_election` and  `/path_to_table/replicas/replica_name/is_active` will be useful.

**Comment by alexey-milovidov** [comment]
Fixed in 20.5-prestable.
#11639
